programs/clearing_house/src/math/margin.rs

```diff
@@ -653,3 +653,58 @@ pub fn calculate_max_withdrawable_amount(
         .safe_mul(precision_increase)?
         .cast()
 }
+
+pub fn validate_spot_margin_trading(
+    user: &User,
+    spot_market_map: &SpotMarketMap,
+    oracle_map: &mut OracleMap,
+) -> ClearingHouseResult {
+    if user.is_margin_trading_enabled {
+        return Ok(());
+    }
+
+    let mut total_open_bids_value = 0_i128;
+    for position_index in 0..user.spot_positions.len() {
+        let asks = user.spot_positions[position_index].open_asks;
+        if asks < 0 {
+            let spot_market =
+                spot_market_map.get_ref(&user.spot_positions[position_index].market_index)?;
+            let signed_token_amount =
+                user.spot_positions[position_index].get_signed_token_amount(&spot_market)?;
+            // The user can have:
+            // 1. no open asks with an existing short
+            // 2. open asks with a larger existing long
+            validate!(
+                signed_token_amount.safe_add(asks.cast()?)? >= 0,
+                ErrorCode::MarginTradingDisabled,
+                "Open asks can lead to increased borrow in spot market {}",
+                user.spot_positions[position_index].market_index
+            )?;
+        }
+
+        let bids = user.spot_positions[position_index].open_bids;
+        if bids > 0 {
+            let spot_market =
+                spot_market_map.get_ref(&user.spot_positions[position_index].market_index)?;
+            let oracle_price_data = oracle_map.get_price_data(&spot_market.oracle)?;
+            let open_bids_value =
+                get_token_value(-bids as i128, spot_market.decimals, oracle_price_data)?;
```

Comment:
does this mean i cant place fully funded buy limit orders $1 below oracle price?

Reply:
yeah this doesnt consider limit prices, similar to how our margin system works